src/ScattBO/utils/ScattBO.py

```diff
@@ -273,6 +273,24 @@ def calculate_loss(x_target, x_sim, Int_target, Int_sim, loss_type="rwp"):
         loss = nn.MSELoss()(Int_target, Int_sim_interp)
     elif loss_type == "smooth_l1":
         loss = nn.SmoothL1Loss()(Int_target, Int_sim_interp)
+    elif loss_type == "shape-metric":
+        # Calculate the shape metric based on: https://github.com/kiranvad/Amplitude-Phase-Distance/tree/main
+        try:
+            from apdist import AmplitudePhaseDistance as dist
+            # Ensure Int_target_double and Int_sim_interp_double are numpy arrays of float64
+            Int_target_double = np.array(Int_target, dtype=np.float64)
+            Int_sim_interp_double = np.array(Int_sim_interp, dtype=np.float64)
+
+            optim_kwargs = {"optim": "DP", "grid_dim": 10}
+            t = (x_target - np.max(x_target)) / (np.max(x_target) - np.min(x_target))
+
+            # Calculate phase_dist and amplitude_dist using the dist function
+            phase_dist, amplitude_dist = dist(t, Int_target_double, Int_sim_interp_double, **optim_kwargs)
+
+            # Convert amplitude_dist to a torch tensor for loss
+            loss = torch.tensor(amplitude_dist, dtype=torch.float32)
+        except ModuleNotFoundError:
+            print("Follow the installation instructions at https://github.com/kiranvad/Amplitude-Phase-Distance/tree/main")
```

Comment:
Hey Andy!

One small question: shouldn't we raise the issue here? Otherwise, there's no loss and the program continues.

I'd maybe change the print statement for a `raise ModuleNotFoundError("Follow the installation instructions at ...")`.